### ISSUE TITLE
RelaxedBernoulli straight through does not contain SVI usage test.

### ISSUE THREAD

**Issue by xidulu** [pyro-ppl/pyro#2629]
Unlike RelaxedOneHotCategorical, which has a test for SVI usage: https://github.com/pyro-ppl/pyro/blob/dev/tests/distributions/test_relaxed_straight_through.py#L60-L78 `RelaxedBernoulli` does not have one.

Follow the test for relaxed categorical, I wrote the following test
```python
pyro.clear_param_store()
ground_truth = 0.8
def model(T):
    p = torch.tensor([ground_truth])
    temp = torch.tensor(T)
    pyro.sample('z', RelaxedBernoulli(temp, p))

def guide(T):
    q = pyro.param('q', torch.tensor([0.4]), constraint=constraints.unit_interval)
    temp = torch.tensor(T)
    pyro.sample('z', RelaxedBernoulli(temperature=temp, probs=q))

adam = optim.Adam({"lr": 0.001, "betas": (0.95, 0.999)})
svi = SVI(model, guide, adam, loss=Trace_ELBO(num_particles=100, vectorize_particles=True))

losses = []
T = 1.0
for k in range(6000):
    loss = svi.step(T)
    T = max(0.5, T * (0.999 ** k))
    losses.append(loss)
    
    
# Some sort of test for atol/rtol    
# assert_almost_equal(ground_truth, pyro.param('q').item(), decimal=3)
```
It managed to pass with a very very high possibility (flakiness is always there, sigh = =)

I could think of a few more tests for RelaxedBernoulli(StraightThourhg) later to verify its correctness and open a PR for it.

**Comment by martinjankowiak** [comment]
from what i recall when this code was written i didn't include such a test precisely because it was really flakey 😬

**Comment by xidulu** [comment]
@martinjankowiak 
The SVI test for one-hot categorical was indeed , very flaky : )
I cannot imagine how could the CI mange to pass this case without manually fix the random seed.
